packages/engine/src/datastore/schema/field_spec/creator.rs

```diff
@@ -2,17 +2,57 @@ use crate::datastore::schema::field_spec::{
     FieldScope, FieldSource, FieldSpec, FieldType, RootFieldSpec,
 };
 
-/// A factory-like object that can be set with a FieldSource and then passed to a context such as a
-/// package. This allows packages to not need to be aware of the FieldSource specifics
+/// Factory for creating `RootFieldSpec`s for a given `FieldSource`.
+///
+/// A `RootFieldSpecCreator` abstracts over a [`FieldSource`] to create [`RootFieldSpec`]s with
+/// a name, a [`FieldType`], and a [`FieldScope`].
 pub struct RootFieldSpecCreator {
     field_source: FieldSource,
 }
 
 impl RootFieldSpecCreator {
-    pub fn new(field_source: FieldSource) -> Self {
+    /// Creates a new `RootFieldSpecCreator` for a given [`FieldSource`].
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use hash_engine::datastore::schema::{FieldSource, RootFieldSpecCreator};
+    ///
+    /// # #[allow(unused_variables)]
```

Comment:
```suggestion
    /// #[allow(unused_variables)]
```

Reply:
This hides the line in the docs. It's only used to hide the warning

Reply:
It hides the line even though it's within the codeblock? That seems very strange. Fair enough though